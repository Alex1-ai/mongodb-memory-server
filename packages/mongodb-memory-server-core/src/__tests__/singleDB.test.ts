/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { MongoClient } from 'mongodb';
import { MongoMemoryServer } from '../index';

jasmine.DEFAULT_TIMEOUT_INTERVAL = 600000;

describe('Single mongoServer', () => {
  let con: MongoClient;
  let mongoServer: MongoMemoryServer;

  beforeAll(async () => {
    mongoServer = await MongoMemoryServer.create();
    con = await MongoClient.connect(mongoServer.getUri(), {
      useNewUrlParser: true,
      useUnifiedTopology: true,
    });
  });

  afterAll(async () => {
    if (con) {
      await con.close();
    }
    if (mongoServer) {
      await mongoServer.stop();
    }
  });

  it('should start mongo server', async () => {
    const db = con.db(mongoServer.instanceInfo!.dbName);

    expect(db).toBeDefined();
    const col = db.collection('test');
    const result = await col.insertMany([{ a: 1 }, { b: 1 }]);
    expect(result.result).toMatchSnapshot();
    expect(await col.countDocuments({})).toBe(2);
  });
});
