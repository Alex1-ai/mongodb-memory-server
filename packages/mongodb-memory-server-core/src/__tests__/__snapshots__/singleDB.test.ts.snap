// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single MongoMemoryServer should successfully set & get information from the database 1`] = `
Object {
  "n": 2,
  "ok": 1,
}
`;
